dask_expr/_expr.py

```diff
@@ -1595,9 +1595,19 @@ def _simplify_up(self, parent, dependents):
     def _lower(self):
         if not isinstance(self, BlockwiseHead):
             # Lower to Blockwise
+            if self.npartitions > self.frame.npartitions:
+                raise ValueError(
+                    f"only {self.frame.npartitions} partitions, head received {self.npartitions}"
+                )
+
+            if isinstance(self, PartitionsFiltered):
+                partitions = self.frame._partitions[: self.operand("npartitions")]
+            else:
+                partitions = list(range(self.frame.npartitions)[: self.npartitions])
```

Comment:
Sorry my recommendation was pretty dumb, you can just do 

```
list(range(self.operand("npartitions"))
```

Reply:
haha, I totally just did that and didn't even register it. :laughing: